DataStructureMemories/DataStructureMemories/DataStructureViewControllers/DetailViewController.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  DetailViewController.swift
+//  DataStructureMemories
+//
+//  Created by Alina Zaitseva on 4/16/18.
+//  Copyright © 2018 Alina Zaitseva. All rights reserved.
+//
+import UIKit
+
+class DetailViewController: UIViewController {
+    
+    @IBOutlet weak var descriptionValue: UILabel!
+    @IBOutlet weak var gradientView: UIView!
+    @IBOutlet weak var halfHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var fullHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var heightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var scrollView: UIScrollView!
+    
+    @IBOutlet weak var buttonToggle: UIButton!
+    @IBOutlet weak var buttonWiki: UIButton!
+    
+    public var srtuctureCell: DataStructMemory?
+    public var factoryEntity = ControlManagerFactory()
+    
+    var isShowingText = false
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = srtuctureCell?.titleOfString() ?? "There is no such title"
+        self.descriptionValue.text = srtuctureCell?.descrOfString() ?? "There is no such description"
+        gradientView.opacityGradient()
+    }
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    @IBAction func buttonShowText(_ sender: Any) {
+        let titleMore = "More"
+        let titleLess = "Less"
+        if isShowingText {
```

Comment:
Better variant, less copy-paste :)

typealias Presentation = (title: String, priority: Int, isGradientHidden: Bool)

var isShowingText: Bool = false {
    didSet {
        self.updatePresentation()
 }
}

var presentation: Presentation {
    return (title: isShowingText ? "Less" : "More", priority: isShowingText ? 999 : 250, isGradientHidden: isShowingText)
}

private func updatePresentation() {
    let presentation = self.presentation
    //update buttonToggle title, constraint and gradientView visibility
}

@IBAction func buttonShowText(_ sender: Any) {
    self. isShowingText = !self. isShowingText
}